third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -17,14 +17,47 @@ namespace ttg = mlir::triton::gpu;
 namespace {
 using tt::DotOp;
 using ttg::AMDMfmaEncodingAttr;
+using ttg::AMDWmmaEncodingAttr;
 using ttg::BlockedEncodingAttr;
 using ttg::ConvertLayoutOp;
 using ttg::DotOperandEncodingAttr;
 using ttg::SliceEncodingAttr;
 
-SmallVector<unsigned, 2>
-warpsPerTileMFMA(tt::DotOp dotOp, const ArrayRef<int64_t> shape, int numWarps,
-                 SmallVector<int64_t, 2> shapePerWarp) {
+enum class MatrixCoreVersion {
+  CDNA_MFMA1,
+  CDNA_MFMA2,
+  CDNA_MFMA3,
+  RDNA_WMMA,
+  UNKNOWN
+};
+
+MatrixCoreVersion getMatrixCoreVersion(StringRef archGen) {
+  if (archGen.contains("gfx11"))
+    return MatrixCoreVersion::RDNA_WMMA;
```

Comment:
is there a reason why we want to pass `archGen` all the way here to decide on matrix core version ? I thought it's enough to use `matrix_core_version` from frontend. (cc @zhanglx13 )

Reply:
RDNA3 arch introduces ``wmma`` instructions that are not technically a new version of matrix core (it is only applicable for CDNA). So it couldn't be described by ``matrix_core_version`` parameter.
Reserving some value for ``wmma`` could be dangerous for future versions of CDNA, thats why I'm passing ``archGen``.
If you have any suggestions please share..

Reply:
One way to unify wmma and mfma in terms of version is to use the gfx number directly, which corresponds to how AMD tracks hw version today and in the future.
And since on AMD path, we don't use capability anyway, we can just use the gfx number as the `version` here:
- 908 <-- matrix_core_version 1
- 90a <-- matrix_core_version 2
- 940, 941, 942 <-- matrix_core_version 3
- 1100 <-- wmma
@zahimoud  what do you think?

Reply:
If we want `int`, we can use 910 for gfx90a.

Reply:
Yea I'm fine with passing arch to backend, I would just pass it to `TargetInfo` object, and in the constructor we can map to an enum class of the arch (probably `MatrixCoreVersion`) and we would have an api like `taretInfo.getMfmaVersion()` to get the mfma version directly.

Reply:
Should we move ``TargetInfo`` from ``TritonAMDGPUToLLVM`` dir to some common place?

Reply:
Let's move it to `amd/include` and do the same for nvidia's backend

Reply:
Well, now it is a part of TritonAMDGPUToLLVM CMake target
I think it would be better to create new target AMDTarget info instead of adding dependency between TritonAMDGPUTransforms and TritonAMDGPUToLLVM ... WDYT?

Reply:
Yea let's move it to amd/include, would that work ?

Reply:
I can see, we also need to move Utility to the separate target. May be there are other nvidia specific pitfalls.
Would it be Ok to do it in separate PR? I can create an issue for that.

Reply:
Which utility are you referring to and why do you think we should move it ?

Reply:
https://github.com/openai/triton/blob/main/third_party/amd/lib/TritonAMDGPUToLLVM/Utility.cpp this one should be moved to separate target to avoid cyclic dependency.

I think it is really need to be done, but this PR is not about it.